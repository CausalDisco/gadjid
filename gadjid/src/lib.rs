// SPDX-License-Identifier: MPL-2.0
#![warn(missing_docs)]
//! gadjid -  Graph Adjustment Identification Distance library
mod ascending_list_utils;
mod graph_loading;
pub mod graph_operations;
mod partially_directed_acyclic_graph;

pub use graph_loading::constructor::EdgelistIterator;
pub use partially_directed_acyclic_graph::LoadError;
pub use partially_directed_acyclic_graph::PDAG;
