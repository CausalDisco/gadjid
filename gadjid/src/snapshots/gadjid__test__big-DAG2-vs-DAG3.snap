---
source: src/lib.rs
expression: "test(&format!(\"100{:0>2}.DAG-100\", true_id),\n    &format!(\"100{:0>2}.DAG-100\", guess_id))"
---
g_true: 10002.DAG-100
g_guess: 10003.DAG-100
ancestor_aid:
  - 0
  - 0
oset_aid:
  - 0.018585858585858588
  - 184
parent_aid:
  - 0.06262626262626263
  - 620
shd:
  - 0.006262626262626263
  - 31
T:
  - 0
  - 1
  - 4
  - 6
  - 10
  - 11
  - 12
  - 16
  - 19
  - 22
  - 23
  - 26
  - 29
  - 30
  - 33
  - 35
  - 43
  - 46
  - 51
  - 52
  - 53
  - 54
  - 55
  - 58
  - 62
  - 65
  - 66
  - 67
  - 69
  - 70
  - 74
  - 77
  - 78
  - 79
  - 81
  - 82
  - 85
  - 87
  - 90
  - 92
  - 93
  - 94
  - 95
  - 96
  - 97
  - 98
NAM: []
Y: 73
random_adj:
  - 2
  - 3
  - 5
  - 7
  - 8
  - 9
  - 13
  - 14
  - 15
  - 17
  - 18
  - 20
  - 21
  - 24
  - 25
  - 27
  - 28
  - 31
  - 32
  - 34
  - 36
  - 37
  - 38
  - 39
  - 40
  - 41
  - 42
  - 44
  - 45
  - 47
  - 48
  - 49
  - 50
  - 56
  - 57
  - 59
  - 60
  - 61
  - 63
  - 64
  - 68
  - 71
  - 72
  - 75
  - 76
  - 80
  - 83
  - 84
  - 86
  - 88
  - 89
  - 91
  - 99
proper_ancestors:
  - 2
  - 3
  - 5
  - 7
  - 8
  - 9
  - 13
  - 14
  - 15
  - 17
  - 18
  - 20
  - 21
  - 24
  - 25
  - 27
  - 28
  - 31
  - 32
  - 34
  - 36
  - 37
  - 38
  - 39
  - 42
  - 45
  - 56
  - 57
  - 64
  - 73
oset:
  - 0
  - 2
  - 4
  - 8
  - 9
  - 12
possible_descendants:
  - 0
  - 1
  - 3
  - 4
  - 5
  - 6
  - 7
  - 10
  - 11
  - 12
  - 13
  - 14
  - 15
  - 16
  - 17
  - 18
  - 19
  - 20
  - 21
  - 22
  - 23
  - 24
  - 25
  - 26
  - 27
  - 28
  - 29
  - 30
  - 31
  - 32
  - 33
  - 34
  - 35
  - 36
  - 37
  - 38
  - 39
  - 40
  - 41
  - 42
  - 43
  - 44
  - 45
  - 46
  - 47
  - 48
  - 49
  - 50
  - 51
  - 52
  - 53
  - 54
  - 55
  - 56
  - 57
  - 58
  - 59
  - 60
  - 61
  - 62
  - 63
  - 64
  - 65
  - 66
  - 67
  - 68
  - 69
  - 70
  - 71
  - 72
  - 73
  - 74
  - 75
  - 76
  - 77
  - 78
  - 79
  - 80
  - 81
  - 82
  - 83
  - 84
  - 85
  - 86
  - 87
  - 88
  - 89
  - 90
  - 91
  - 92
  - 93
  - 94
  - 95
  - 96
  - 97
  - 98
  - 99
parent_adjustment_NVA:
  - 0
  - 1
  - 2
  - 3
  - 4
  - 5
  - 6
  - 7
  - 8
  - 9
  - 10
  - 11
  - 12
  - 13
  - 14
  - 15
  - 16
  - 17
  - 18
  - 19
  - 20
  - 21
  - 22
  - 23
  - 24
  - 25
  - 26
  - 27
  - 28
  - 29
  - 30
  - 31
  - 32
  - 33
  - 34
  - 35
  - 36
  - 37
  - 38
  - 39
  - 40
  - 41
  - 42
  - 43
  - 44
  - 45
  - 46
  - 47
  - 48
  - 49
  - 50
  - 51
  - 52
  - 53
  - 54
  - 55
  - 56
  - 57
  - 58
  - 59
  - 60
  - 61
  - 62
  - 63
  - 64
  - 65
  - 66
  - 67
  - 68
  - 69
  - 70
  - 71
  - 72
  - 73
  - 74
  - 75
  - 76
  - 77
  - 78
  - 79
  - 80
  - 82
  - 83
  - 84
  - 85
  - 86
  - 87
  - 88
  - 89
  - 90
  - 91
  - 92
  - 96
  - 97
  - 99
ancestor_adjustment_NVA:
  - 0
  - 1
  - 2
  - 3
  - 4
  - 5
  - 6
  - 7
  - 8
  - 9
  - 10
  - 11
  - 12
  - 13
  - 14
  - 15
  - 16
  - 17
  - 18
  - 19
  - 20
  - 21
  - 22
  - 23
  - 24
  - 25
  - 26
  - 27
  - 28
  - 29
  - 30
  - 31
  - 32
  - 33
  - 34
  - 35
  - 36
  - 37
  - 38
  - 39
  - 40
  - 41
  - 42
  - 43
  - 44
  - 45
  - 46
  - 47
  - 48
  - 49
  - 50
  - 51
  - 52
  - 53
  - 54
  - 55
  - 56
  - 57
  - 58
  - 59
  - 60
  - 61
  - 62
  - 63
  - 64
  - 65
  - 66
  - 67
  - 68
  - 69
  - 70
  - 71
  - 72
  - 73
  - 74
  - 75
  - 76
  - 77
  - 78
  - 79
  - 80
  - 81
  - 82
  - 83
  - 84
  - 85
  - 86
  - 87
  - 88
  - 89
  - 90
  - 91
  - 92
  - 96
  - 97
  - 99
nondescendant_adjustment_NVA:
  - 2
  - 3
  - 5
  - 7
  - 8
  - 9
  - 13
  - 14
  - 15
  - 17
  - 18
  - 20
  - 21
  - 24
  - 25
  - 27
  - 28
  - 31
  - 32
  - 34
  - 36
  - 37
  - 38
  - 39
  - 40
  - 41
  - 42
  - 44
  - 45
  - 47
  - 48
  - 49
  - 50
  - 56
  - 57
  - 59
  - 60
  - 61
  - 63
  - 64
  - 68
  - 71
  - 72
  - 73
  - 75
  - 76
  - 80
  - 83
  - 84
  - 86
  - 88
  - 89
  - 91
  - 99
oset_adjustment_NVA:
  - 0
  - 2
  - 3
  - 4
  - 5
  - 7
  - 8
  - 9
  - 12
  - 13
  - 14
  - 15
  - 17
  - 18
  - 20
  - 21
  - 24
  - 25
  - 27
  - 28
  - 31
  - 32
  - 34
  - 36
  - 37
  - 38
  - 39
  - 40
  - 41
  - 42
  - 44
  - 45
  - 47
  - 48
  - 49
  - 50
  - 56
  - 57
  - 59
  - 60
  - 61
  - 63
  - 64
  - 68
  - 71
  - 72
  - 73
  - 75
  - 76
  - 80
  - 83
  - 84
  - 86
  - 88
  - 89
  - 91
  - 99
empty_adjustment_NVA:
  - 2
  - 3
  - 5
  - 7
  - 8
  - 9
  - 13
  - 14
  - 15
  - 17
  - 18
  - 20
  - 21
  - 24
  - 25
  - 27
  - 28
  - 31
  - 32
  - 34
  - 36
  - 37
  - 38
  - 39
  - 40
  - 41
  - 42
  - 44
  - 45
  - 47
  - 48
  - 49
  - 50
  - 56
  - 57
  - 59
  - 60
  - 61
  - 63
  - 64
  - 68
  - 71
  - 72
  - 73
  - 75
  - 76
  - 80
  - 83
  - 84
  - 86
  - 88
  - 89
  - 91
  - 99
random_Z_adjustment_NVA:
  - 2
  - 3
  - 5
  - 7
  - 8
  - 9
  - 13
  - 14
  - 15
  - 17
  - 18
  - 20
  - 21
  - 24
  - 25
  - 27
  - 28
  - 31
  - 32
  - 34
  - 36
  - 37
  - 38
  - 39
  - 40
  - 41
  - 42
  - 44
  - 45
  - 47
  - 48
  - 49
  - 50
  - 56
  - 57
  - 59
  - 60
  - 61
  - 63
  - 64
  - 68
  - 71
  - 72
  - 73
  - 75
  - 76
  - 80
  - 83
  - 84
  - 86
  - 88
  - 89
  - 91
  - 99
